perun-rpc/src/main/java/cz/metacentrum/perun/rpc/csrf/CsrfFilter.java

```diff
@@ -116,7 +116,9 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 				log.trace("Invalid CSRF token found for {} COOKIE: {} | HEADER: {}.", request.getRequestURI(), cookieCsrfToken.getValue(), actualToken);
 
 				// token is either invalid or missing
-				response.setStatus(HttpStatus.SC_FORBIDDEN);
+				response.sendError(HttpStatus.SC_FORBIDDEN, "This domain is under CSRF protection; however, " +
+				    "a valid CSRF token was not provided. Configure the client to provide the token in header " +
+					"or use the domain reserved for API calls - without CSRF protection");
```

Comment:
Just a minor suggestion - I think this would be a little more accurate. What do you think?
```suggestion
					"or use the domain reserved only for API calls - without CSRF protection");
```